tsl/src/nodes/vector_agg/plan.c

```diff
@@ -177,6 +180,19 @@ vector_agg_plan_create(Agg *agg, CustomScan *decompress_chunk, List *resolved_ta
 	lfirst(list_nth_cell(vector_agg->custom_private, VASI_GroupingType)) =
 		makeInteger(grouping_type);
 
+#if PG15_GE
+	if (is_columnar_scan(custom))
+	{
+		/*
+		 * ColumnarScan should not project when doing vectorized
+		 * aggregation. If it projects, it will turn the arrow slot into a set
+		 * of virtual slots and the vector data will not be passed up to
+		 * VectorAgg.
+		 */
+		custom->flags &= ~CUSTOMPATH_SUPPORT_PROJECTION;
```

Comment:
Does this actually work? Since we're injecting the vector agg plan in the post-planning hook, it might be too late for this.

Reply:
PostgreSQL still plans projection, but in our own code we explicitly checks whether this flag is later turned off when executing ColumnarScan, and in that case we skip projection. So, it is a custom check by us where we reuse this flag. We could also use a separate "boolean" to signal that we should skip projection, but this was a convenient flag to use given that it already deals with projection.

Reply:
I can clarify in the comment if you think that would help.

Reply:
> I can clarify in the comment if you think that would help.

Yes, a comment would be helpful, this confused me at first.

Reply:
Expanded on the comment.